src/views/enroll-factors/BarcodeView.js

```diff
@@ -44,9 +44,25 @@ define([
     events: {
       'click [data-type="manual-setup"]': function (e) {
         e.preventDefault();
-        var url = RouterUtil.createActivateFactorUrl(this.model.get('__provider__'),
+        const url = RouterUtil.createActivateFactorUrl(this.model.get('__provider__'),
           this.model.get('__factorType__'), 'manual');
-        this.options.appState.trigger('navigate', url);
+        if (this.model.get('__factorType__') === 'push') {
+          // cancel the poll and navigate to manual setup.
+          this.model.doTransaction(function (transaction) {
+            return transaction.prev()
+              .then(function (trans) {
+                var factor = _.findWhere(trans.factors, {
```

Comment:
nit: It'd be useful for the future to add a comment here explaining why we're doing this - reading the code, I don't think it'd be obvious why we're in the push state, cancel it, and then go back to it.